common/djangoapps/enrollment/views.py

```diff
@@ -301,6 +302,81 @@ def get(self, request, course_id=None):
             )
 
 
+class UnenrollmentView(APIView):
+    """
+        **Use Cases**
+
+            * Unenroll a single user from all courses.
+
+              This command should only be issued from a Jenkins job that retires a user
+              from edX.
+
+        **Example Requests**
+
+            POST /api/enrollment/v1/enrollment {
+                "user": "username12345"
+            }
+
+            **POST Parameters**
+
+              A POST request must include the following parameter.
+
+              * user: The username of the user being unenrolled.
+              This will never match the username from the request, since
+              the request is issued via Jenkins.
+
+        **POST Response Values**
+
+             If the user does not exist, or the user is already unenrolled
+             from all courses, the request returns an HTTP 404 "Does Not Exist"
+             response.
+
+             If an unexpected error occurs, the request returns an HTTP 500 response.
+
+            If the request is successful, an HTTP 200 "OK" response is
+            returned along with a list of all courses from which the user was unenrolled.
+        """
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, CanRetireUser)
+
+    def post(self, request):
+        """
+        Unenrolls the specified user from all courses.
+        """
+        # Get the User from the request.
+        username = request.data.get('user', None)
+        if not username:
```

Comment:
Do we need this check? If the username is `None`, the `User` model will throw in the try/except below. If it throws something other than `ObjectDoesNotExist`, it can be handled as an additional except clause.

Reply:
We could bundle this into the `DoesNotExist` check further down. I broke this case out because I think of "the user doesn't exist" and "you didn't say which user to unenroll" as usefully different messages, and the plan for all these endpoints is that they provide as detailed a response as possible.

Reply:
Got it. I agree: they are separate failures worth communicating. 

You may be able to handle both errors as separate `catch` clauses in the `try/catch`. Checking for a `username` up front is easier to reason about. My comment is nit-picky, so feel free to disregard.